src/_base/harness/scripts/enable.sh.twig

```diff
@@ -2,10 +2,23 @@
 
 main()
 {
-    passthru ws networks external
-    if [ ! -f .my127ws/.flag-built ]; then
+    local IS_BUILT=no
+    local HAS_FLAG=no
+
+    if [ -f .my127ws/.flag-built ]; then
+        HAS_FLAG=yes
+        IS_BUILT=yes
+        # If no lines for `docker-compose ps`, the containers no longer exist so we need to rebuild them
+        if [ "$(docker-compose ps --quiet | wc -l | awk '{ print $1 }')" = "0" ]; then
+            IS_BUILT=no
+        fi
+    fi
 
-        passthru docker-compose down
+    passthru ws networks external
+    if [ "$IS_BUILT" = "no" ]; then
+        if [ "$HAS_FLAG" = "no" ]; then
+          passthru docker-compose down
+        fi
```

Comment:
So if the setup process is halted by something, it would _retry_ the setup from where it stopped instead of removing everything and starting from scratch?

Reply:
If the setup process stops, the flag is never created, so would still bring containers down to try again.

Reply:
ah right, I was thinking about an existing env, where I retry the setup and it stops, but then the retry step would/should have removed the flag, all good 👍